apiserver/facades/agent/uniter/state.go

```diff
@@ -10,45 +10,79 @@ import (
 	"github.com/juju/juju/state"
 )
 
-type storageStateInterface interface {
-	RemoveStorageAttachment(names.StorageTag, names.UnitTag) error
+type storageInterface interface {
 	StorageInstance(names.StorageTag) (state.StorageInstance, error)
-	StorageInstanceFilesystem(names.StorageTag) (state.Filesystem, error)
-	StorageInstanceVolume(names.StorageTag) (state.Volume, error)
 	UnitStorageAttachments(names.UnitTag) ([]state.StorageAttachment, error)
+	RemoveStorageAttachment(names.StorageTag, names.UnitTag) error
 	DestroyUnitStorageAttachments(names.UnitTag) error
 	StorageAttachment(names.StorageTag, names.UnitTag) (state.StorageAttachment, error)
-	UnitAssignedMachine(names.UnitTag) (names.MachineTag, error)
-	FilesystemAttachment(names.MachineTag, names.FilesystemTag) (state.FilesystemAttachment, error)
-	VolumeAttachment(names.MachineTag, names.VolumeTag) (state.VolumeAttachment, error)
+	AddStorageForUnit(tag names.UnitTag, name string, cons state.StorageConstraints) ([]names.StorageTag, error)
 	WatchStorageAttachments(names.UnitTag) state.StringsWatcher
 	WatchStorageAttachment(names.StorageTag, names.UnitTag) state.NotifyWatcher
-	WatchFilesystemAttachment(names.MachineTag, names.FilesystemTag) state.NotifyWatcher
+}
+
+type storageVolumeInterface interface {
+	StorageInstanceVolume(names.StorageTag) (state.Volume, error)
+	BlockDevices(names.MachineTag) ([]state.BlockDeviceInfo, error)
 	WatchVolumeAttachment(names.MachineTag, names.VolumeTag) state.NotifyWatcher
 	WatchBlockDevices(names.MachineTag) state.NotifyWatcher
-	AddStorageForUnit(tag names.UnitTag, name string, cons state.StorageConstraints) ([]names.StorageTag, error)
-	UnitStorageConstraints(u names.UnitTag) (map[string]state.StorageConstraints, error)
-	BlockDevices(names.MachineTag) ([]state.BlockDeviceInfo, error)
+	VolumeAttachment(names.MachineTag, names.VolumeTag) (state.VolumeAttachment, error)
 }
 
-type storageStateShim struct {
-	*state.State
-	*state.IAASModel
+type storageFilesystemInterface interface {
+	StorageInstanceFilesystem(names.StorageTag) (state.Filesystem, error)
+	FilesystemAttachment(names.MachineTag, names.FilesystemTag) (state.FilesystemAttachment, error)
+	WatchFilesystemAttachment(names.MachineTag, names.FilesystemTag) state.NotifyWatcher
 }
 
-var getStorageState = func(st *state.State) (storageStateInterface, error) {
-	im, err := st.IAASModel()
+var getStorageState = func(st *state.State) (storageInterface, storageVolumeInterface, storageFilesystemInterface, error) {
```

Comment:
I think my suggestion would tidy this up too - you'd return one thing + error from here rather than four.

Reply:
Yep, fixed this